### ISSUE TITLE
Call for Contributors: A Cloud App for Solar Panel Spotting!

### ISSUE THREAD

**Issue by flowirtz** [openclimatefix-archives/bigsolarhunt#7]
[Open Climate Fix](https://openclimatefix.org/) is collaborating with [Possible](https://www.wearepossible.org/), the UK charity that inspires climate action. We are building a phone app for members of the public to help spot solar panels in the street. We're now looking for a climate-minded coder who can volunteer their skills and a small time commitment, to bring this app into production.

The app has already been designed and built (using Dart/Flutter). The work to be done involves connecting the app to a cloud backend for storing/uploading images, and then to test the app before public release in spring/summer this year.

*Why do this?* [We aim to map all of the million+ solar PV installations in the UK](https://openclimatefix.org/blog/2019-07-09-solar-pv-mapping), using crowdsourcing and machine learning. This will improve the country's solar PV power predictions. This phone app will directly contribute to this goal, plus it will create an open training dataset which will also help other countries to do the same thing worldwide.

## Recommended Skills
- Cloud storage APIs (e.g. AWS);
- Mobile app development using [Flutter](https://flutter.dev/)

## Time Estimate
2 hours a week for 3 months, according to availability.

## Next Steps
- [ ] https://github.com/openclimatefix/bigsolarhunt/issues/5
- [ ] https://github.com/openclimatefix/bigsolarhunt/issues/6

## Contact
Just reply to this issue! 😊 

## Screenshots of the App
The screenshots below show features that are already implemented.

<img src="https://user-images.githubusercontent.com/6052785/148976666-4c6b6e4b-ec37-4d08-a831-44a79a4f45f3.png" width="200" height="400" />
<img src="https://user-images.githubusercontent.com/6052785/148976676-263b8850-a88b-4713-92c2-28431e3a4716.png" width="200" height="400" />
<img src="https://user-images.githubusercontent.com/6052785/148976680-b29adafc-90b8-4e29-9998-16646d69bfa7.png" width="200" height="400" />

**Comment by mikejamesthompson** [comment]
Are you still looking for people to help with this?

**Comment by danstowell** [comment]
@mikejamesthompson Yes! This project has been put to a low priority, because it seems unlikely we'll have an app fully ready for the northern-hemisphere spring/summer 2022. But if you're willing+able to look into it, please do make a start, look into the codebase and the issues, and we can discuss it.

**Comment by mikejamesthompson** [comment]
> @mikejamesthompson Yes! This project has been put to a low priority, because it seems unlikely we'll have an app fully ready for the northern-hemisphere spring/summer 2022. But if you're willing+able to look into it, please do make a start, look into the codebase and the issues, and we can discuss it.

Thanks @danstowell. I've been poking around and have the app running locally. I have Flutter 2.10.5 though, and it complained about various things including Android v2 embedding not being quite right, some deprecated APIs and a few other bits.

I have a branch here with some commits to fix these things. Is there anyone on your side who could review this if I submit?

On AWS integration, I will comment on that on the separate issue.

**Comment by jacobbieker** [comment]
Hi,

I can do a code review if you want, it's been a bit since I did Android development, but can definitely check it out! Thanks for the work!
